packages/Webkul/Admin/src/Http/Controllers/Catalog/ProductController.php

```diff
@@ -382,4 +382,21 @@ public function checkVariantUniqueness(): JsonResponse
 
         return new JsonResponse([]);
     }
+
+    public function checkVariantValues(): JsonResponse
+    {
+        $variantAttributes = request()->get('variantAttributes');
+
+        foreach ($variantAttributes as $variantAttributeCode => $variantAttributeValue) {
+            if (empty($variantAttributeValue)) {
+                return new JsonResponse([
+                    'errors' => [
+                        'message' => trans('admin::app.catalog.products.edit.types.configurable.variant-attribute-option-not-selected', ['attribute' => $variantAttributeCode]),
+                    ],
+                ]);
+            }
+        }
+
+        return new JsonResponse([]);
+    }
```

Comment:
This function is only checking for empty values which could have been done in js without an ajax call.